uiplib/scheduler.py

```diff
@@ -2,79 +2,64 @@
 
 import os
 import time
-import sys
 from select import select
 import random
 
 from uiplib.utils.utils import get_percentage
-from uiplib.scrape import download
+from uiplib.scrape import get_image_data_list, download_and_store_image
 
 try:
     import msvcrt
 except ImportError:
     # not on windows
     pass
-from threading import Thread
 
 
 class Scheduler:
     """Class which schedules the wallpaper change."""
 
-    def __init__(self, offline, pics_folder, timeout, website, count,
-                 skip_wallpaper, wallpaper, appObj=None):
+    def __init__(self, pics_folder, timeout, website, wallpaper):
         """Initialize the scheduler configuration."""
         self.website = website
         self.timeout = timeout
         self.directory = pics_folder
-        self.count = count
-        self.skip_wallpaper = skip_wallpaper
         self.wallpaper = wallpaper
-        self.offline = offline
-        self.appObj = appObj
-        self.download_thread = None
         self.schedule_thread = None
+        self.image_data_list = []
+        print('all fine till here')
         self.run()
 
-    def change_random(self):
-        """Change the wallpaper to a random image."""
-        filename = random.choice(os.listdir(self.directory))
-        path = os.path.join(self.directory, filename)
-        print("changing desktop wallpaper to: ", path)
-        self.wallpaper.set(path)
-
-    def kbhit(self):
-        """Return True if keyboard character was hit, False otherwise."""
-        if not self.skip_wallpaper:
-            return False
-        if os.name == 'nt':
-            return msvcrt.kbhit()
-        else:
-            dr, dw, de = select([sys.stdin], [], [], 0)
-            return dr != []
-
-    def getch(self):
-        """Return a keyboard character after kbhit() has been called.
-
-        Should not be called in the same program as getarrow().
-        """
-        if os.name == 'nt':
-            return msvcrt.getch().decode('utf-8')
+    def change_next(self):
+        """Change the wallpaper to the next image."""
+        prev_wallpaper_path = self.wallpaper.get()[0]
+        next_wallpaper_data = self.get_next_wallpaper()
+        download_and_store_image(self.directory,
+                                 next_wallpaper_data)
+        path = os.path.join(self.directory,
+                            next_wallpaper_data.get('name'))
+        if os.path.exists(path):
+            self.wallpaper.set(path)
+        if os.path.exists(prev_wallpaper_path):
+            os.remove(prev_wallpaper_path)
```

Comment:
Uhm. Deleting the previous one? But why! It’s already been downloaded, right?

Reply:
Not deleting it leads to various other code like autoflush, and why have that when you can avoid it at creation?